### ISSUE TITLE
"can be inherently insecure" in Windows setup is unclear and not explained

### ISSUE THREAD

**Issue by matkoniecz** [ansible/ansible#72449]
##### SUMMARY

> The ConfigureRemotingForAnsible.ps1 script is intended for training and development purposes only and should not be used in a production environment, since it enables settings (like Basic authentication) that can be inherently insecure.

It is not making clear what kind of security issue is created. In my case I am using Ansible to setup my own private laptop and I am unsure whatever this kind of use falls under "training and development purposes" (Ansible changing things from within WSL on the Windows system).

Or maybe by running this script anyone on the internet may connect to my laptop and take over it?

> it enables settings (like Basic authentication) that can be inherently insecure.

Is not distinguishing between "unsuitable for managing fleet of 7272727 servers, not a problem for single-device Ansible use" and "run it on computer connected to internet and it will become spambot with 15 minutes, use only in VMs never on real devices"

https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/user_guide/windows_setup.rst

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME

windows_setup.rst

##### ANSIBLE VERSION

```paste below
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/mateusz/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

Not actually relevant, as I am reporting issue in docs from devel branch.

##### CONFIGURATION

```paste below
INTERPRETER_PYTHON(/etc/ansible/ansible.cfg) = auto
```

Not actually relevant, as I am reporting issue in docs from devel branch.

##### OS / ENVIRONMENT

None is relevant, as it is about unclear docs

##### ADDITIONAL INFORMATION

Sadly I am unsure whatever to describe it as a something problematic while deploying and controlling other devices over network or something insecure in any case.

**Comment by ansibot** [comment]
Files identified in the description:
* [`docs/docsite/rst/user_guide/windows_setup.rst`](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/user_guide/windows_setup.rst)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
There are a few issues with this script

* It enables Basic auth
    * Windows has this disabled by default because it is not secure when running over HTTP
    * Over HTTP this is an issue because the messages are not encrypted, even the credentials are just base64 encoded
    * Anyone sniffing the network will be able to get your credentials as well as any data that Ansible sends to the remote host and what it receives back
    * Requires HTTPS to be set up to be secure, even then it's problematic because it only supports local account and not domain accounts
* It creates a self signed certificate
    * Fine for basic testing but in a domain environment you should be using your own CA issued certificates for the host
    * Also it creates both a HTTP and HTTPS listener, where only the latter is enabled by default. Not really an issue but some places like to reduce the amount of inbound entrypoints as much as they can
* It allows WinRM traffic in all network profiles
    * Usually you want to restrict traffic to just the network/profile you will be managing it from

A lot of these settings are done (Basic auth and HTTPS with self signed certificate) because it was created in a time where the WinRM support on Python was very basic and barebones. There was no NTLM or Kerberos support, no message encryption over HTTP. This is no longer the case and the Python WinRM library that is used supports the full gauntlet of authentication protocols WinRM supports as well as message encryption when it's run over HTTP. In fact, Ansible has been able to connect to a Windows host that has already run `Enable-PSRemoting` in PowerShell already. There's really no need to use this script at all as the defaults in Windows are just fine.

The reason why we still have this script is simple, people still use it and have their scripts set to download it directly from https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1. If we were to remove it, or even just some of the default behaviour to be a bit more secure we will be breaking plenty of scripts that still rely on its current behaviour.

What we should probably do is add a comment to try and discourage the use of this script even more but even then we can't really delete it until we are ready to face a backlash of users who rely on it currently.

As for your question it's really a matter of the network and whether you are a member of a domain.

* If you are connecting to domain joined servers then you are perfectly fine connecting with Kerberos authentication
* If you are connecting over a public network then you should be using HTTPS with `ntlm` auth
    * It's probably preferable to use your own CA signed certificates to ensure you are connecting to the right hosts but you can definitely extract the self signed cert setup part of the configure code and run just that

TLDR: We are trying to discourage users from using the script as much as we can. It was created for a time where Ansible couldn't work with the default WinRM settings. If you do wish to use it then take a copy of the script and only run the parts you need, i.e. self signed certificates and so on.

**Comment by matkoniecz** [comment]
> The reason why we still have this script is simple, people still use it and have their scripts set to download it directly from https://github.com/ansible/ansible/blob/devel/examples/scripts/ConfigureRemotingForAnsible.ps1.

Still hosting script is fine, but why it is listed as recommended step in setting up Ansible on Windows if it is not needed anymore?

> We are trying to discourage users from using the script as much as we can.

Maybe keep hosting but remove it from https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/user_guide/windows_setup.rst#winrm-setup ? Maybe keep note about old, no longer needed script hosted for backward compatibility without mentioning running it as an useful step.

If that would be desirable it is a PR that I can do.

Right now "We are trying to discourage users from using the script as much as we can." is not true, as new user I was looking for a way run Ansible from WSL and I found no other setup guide. OK, I found setup guides that did everything what this setup script did and enabled also other even more questionable options.

>  then take a copy of the script and only run the parts you need, i.e. self signed certificates and so on.

Maybe have an old script at current location for backward compatibility and create new one with only still needed parts?

> Windows has this disabled by default because it is not secure when running over HTTP

I am still unsure is it OK to enable it for test Ansible run from within WSL. Is it going to open some security holes exploitable in this context, in internal connection going from WSL to Windows on the same laptop? Is it affecting any other software on the laptop? Is it exploitable remotely from Internet?

Is it "unsuitable for managing fleet of 7272727 servers, not a problem for single-device Ansible use" or "run it on computer connected to internet and it will become spambot with 15 minutes, use only in VMs never on real devices"?
